File: Programming Basics/Programming Basics - C#/Exams/Programming Basics Exam - 17 July 2016/Exam - 17 July 2016/2. Harvest/Harvest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _2.Harvest
{
    class Harvest
    {
        static void Main(string[] args)
        {
            int X = int.Parse(Console.ReadLine());
            double Y = double.Parse(Console.ReadLine());
            int Z = int.Parse(Console.ReadLine());
            int workers = int.Parse(Console.ReadLine());

            double percentage = X * 0.40;
            double kg = percentage * Y;
            double wineLiters = kg / 2.5;

            if (wineLiters < Z)
            {
                double wineLacked = Math.Floor(Z - wineLiters);
                Console.WriteLine("It will be a tough winter! More {0:0} liters wine needed.", wineLacked);
            }

            else
            {
                Console.WriteLine("Good harvest this year! Total wine: {0} liters.", Math.Floor(wineLiters));
                double wineLeft = Math.Ceiling(wineLiters - Z);
                double winePerPerson = Math.Ceiling(wineLeft / workers);

                Console.WriteLine("{0} liters left -> {1} liters per person.", wineLeft, winePerPerson);
            }
        }
    }
}

File: Programming Basics/Programming Basics - JavaScript/Exercises/03. Simple Conditions/20. Harvest/Harvest.js
function calculateTheWine([arg1, arg2, arg3, arg4]) {
    let area = parseFloat(arg1);
    let grapes = parseFloat(arg2);
    let litersNeeded = Number(arg3);
    let workers = Number(arg4);

    let harvestForWine = area * 0.40;
    let khGrapes = grapes * harvestForWine;
    let litersWine = khGrapes / 2.5;

    if (litersWine < litersNeeded) {
        console.log(`It will be a tough winter! More ${Math.floor(litersNeeded - litersWine)} liters wine needed.`);
    }

    else {
        console.log(`Good harvest this year! Total wine: ${Math.floor(litersWine)} liters.`);
        let litersWineLeft = Math.ceil(litersWine - litersNeeded);
        let litersWinePerWorker = Math.ceil(litersWineLeft / workers);
        console.log(`${litersWineLeft} liters left -> ${litersWinePerWorker} liters per person.`);
    }
}

/* Uncomment to test:
calculateTheWine(["650", "2", "175", "3"]);
calculateTheWine(["1020", "1.5", "425", "4"]);
    */